### PR TITLE
Add functions to store and retrieve map frames in the database

### PR DESCRIPTION
**Pull request by mhhd2020**

### CODE REVIEW

File: sketch_map_tool/database/client.py [hunk 1]
```diff
@@ -54,8 +57,7 @@ def _select_id_map(uuid) -> dict:
         raw = curs.fetchall()
     if raw:
         return raw[0][0]
-    else:
-        raise UUIDNotFoundError("There are no tasks in the broker for UUID: " + uuid)
+    raise UUIDNotFoundError("There are no tasks in the broker for UUID: " + uuid)
```
Comment (matthiasschaub) [inline sketch_map_tool/database/client.py:60]: why did you remove the else statement?

File: sketch_map_tool/database/client.py [hunk 2]
```diff
@@ -86,10 +88,9 @@ def _select_file(id_: int) -> memoryview:
         raw = curs.fetchone()
         if raw:
             return raw[0]
-        else:
-            raise FileNotFoundError_(
-                "There is no file in the database with the id: " + str(id_)
-            )
+        raise FileNotFoundError_(
```
Comment (matthiasschaub) [inline sketch_map_tool/database/client.py:91]: see above

File: sketch_map_tool/database/client.py [hunk 3]
```diff
@@ -138,3 +139,42 @@ def read_file(id_: int) -> memoryview:
...
+    """Write map_frame image for georeferencing to the database under a given UUID"""
+    create_query = """
+    CREATE TABLE IF NOT EXISTS map_frame(
+        uuid uuid PRIMARY KEY,
+        file BYTEA
+        )
+        """
+    insert_query = "INSERT INTO map_frame(uuid, file) VALUES (%s, %s)"
+    with db_conn.cursor() as curs:
+        # executemany and fetchall does not work together
```
Comment (matthiasschaub) [inline sketch_map_tool/database/client.py:172]:
```suggestion
```
not needed

File: sketch_map_tool/database/client.py (hunk 1 above)
Comment (mhhd2020) [reply sketch_map_tool/database/client.py]: For style reasons, restored previous version

**Review by matthiasschaub — COMMENTED** [review]
Looks good. Just some small comments.

File: sketch_map_tool/database/client.py [hunk 4]
```diff
@@ -141,6 +143,45 @@ def _delete_file(id_: int):
...
+    query = "SELECT file FROM map_frame WHERE uuid = %s"
+    with db_conn.cursor() as curs:
+        curs.execute(query, [str(uuid)])
+        raw = curs.fetchone()
+        if raw:
+            return raw[0]
+        else:
+            raise FileNotFoundError_(
+                f"There is no file in the database table 'map_frame' with the uuid: {uuid}"
+            )
```
Comment (joker234) [inline sketch_map_tool/database/client.py:175]:
```suggestion
            raise FileNotFoundError_(
                "There is no map frame in the database with the uuid: {}".format(uuid)
            )
```

File: sketch_map_tool/database/client.py [hunk 5]
```diff
@@ -141,6 +143,45 @@ def _delete_file(id_: int):
...
 
+def _insert_map_frame(file: BytesIO, uuid: UUID):
+    """Insert map frame as blob into the database with the uuid as primary key.
+
+    The map frame is later on needed for georeferencing the uploaded photo or scane of
+    a sketch map.
+    """
+    create_query = """
+    CREATE TABLE IF NOT EXISTS map_frame(
+        uuid uuid PRIMARY KEY,
```
Comment (joker234) [inline sketch_map_tool/database/client.py:154]:
```suggestion
        uuid UUID PRIMARY KEY,
```

File: sketch_map_tool/database/client.py [hunk 6]
```diff
@@ -141,6 +143,45 @@ def _delete_file(id_: int):
         curs.execute(query, [id_])
 
 
+def _insert_map_frame(file: BytesIO, uuid: UUID):
+    """Insert map frame as blob into the database with the uuid as primary key.
+
+    The map frame is later on needed for georeferencing the uploaded photo or scane of
+    a sketch map.
```
Comment (joker234) [inline sketch_map_tool/database/client.py:150]:
```suggestion
    The map frame is later on needed for georeferencing the uploaded photo or scan of
    a sketch map.
```

**Review by joker234 — DISMISSED** [review]

**Review by joker234 — APPROVED** [review]

**Review by joker234 — APPROVED** [review]
